.github/workflows/tests-pr.yml

```diff
@@ -0,0 +1,54 @@
+name: BNB tests on Pull Requests
+
+on:
+  workflow_dispatch:
+    inputs:
+      pr_number:
+        description: 'PR number'
+        required: true 
+        
+env:
+  # To be able to run tests on CUDA 12.2
+  NVIDIA_DISABLE_REQUIRE: "1"
+  SLACK_API_TOKEN: ${{ secrets.SLACK_API_TOKEN }}
+  OUTPUT_SLACK_CHANNEL_ID: "C071049S8EM"
+
+jobs:
+  run_tests_single_gpu:
+    strategy:
+      fail-fast: false
+    runs-on: [self-hosted, single-gpu, nvidia-gpu, t4, ci]
+    container:
+      image: "huggingface/peft-gpu-bnb-latest:latest"
```

Comment:
we use this image that contains all HF libs installed from latest pypi, we already test bnb on HF libs from source on daily basis